Explorer/Assets/DCL/AvatarRendering/Wearables/Helpers/WearableCatalog.cs

```diff
@@ -1,47 +1,111 @@
 ﻿using DCL.AvatarRendering.Wearables.Components;
+using DCL.PerformanceAndDiagnostics.Optimization.PerformanceBudgeting;
 using ECS.StreamableLoading.Common.Components;
+using System;
 using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Pool;
 
 namespace DCL.AvatarRendering.Wearables.Helpers
 {
-    public class WearableCatalog
+    public class WearableCatalog : IDisposable
     {
-        internal Dictionary<string, IWearable> wearableDictionary;
+        private static readonly Comparison<(string key, uint lastUsedFrame)> compareByLastUsedFrame =
+            (pair1, pair2) => pair1.lastUsedFrame.CompareTo(pair2.lastUsedFrame);
 
-        public WearableCatalog()
+        private readonly List<(string key, uint lastUsedFrame)> listedCacheKeys = ListPool<(string, uint)>.Get();
```

Comment:
I think this collection does not suit use scenarios well:

- `FindIndex` is linear and the list can be very large
- And Resorting is quite expensive as well